src/app/onion/dashboard/dashboard.component.ts

```diff
@@ -1,15 +1,51 @@
 import { Component, OnInit } from '@angular/core';
+import { HistoryService } from 'app/core/history.service';
+import { NavigationEnd, Router } from '@angular/router';
+import { NavbarService } from 'app/core/navbar.service';
+import { Subject } from 'rxjs';
 
 @Component({
   selector: 'clark-dashboard',
   templateUrl: './dashboard.component.html',
-  styleUrls: ['./dashboard.component.scss']
+  styleUrls: ['./dashboard.component.scss'],
 })
-export class DashboardComponent implements OnInit {
+export class DashboardComponent {
+  lastLocation: NavigationEnd;
+  activeIndex = 0;
 
-  constructor() { }
+  action$: Subject<string> = new Subject();
 
-  ngOnInit() {
+  constructor(
+    private history: HistoryService,
+    private router: Router,
+    private navbar: NavbarService
+  ) {
+    this.navbar.hide();
+    this.lastLocation = this.history.lastRoute;
   }
 
+  /**
+   * Toggles between the draft tab and the released tab
+   */
+  toggle() {
+    if (this.activeIndex % 2) {
+      this.action$.next('-1');
+    } else {
+      this.action$.next('+1');
```

Comment:
What is the purpose of using a string form of a numeric value here?

Reply:
To be completely honest, I have no idea why I set the carousel component up like that. It makes more sense to be numeric